src/auth.js

```diff
@@ -25,7 +25,7 @@ async function retrieveToken(method, client) {
         case 'jwt': {
             /** @type {string} */
             let jwt;
-            const role = core.getInput('role', { required: true });
+            const role = core.getInput('role', { required: false });
```

Comment:
I think you'll also need to exclude passing `role` to `getClientToken` below on line 41 if it's not set.

Reply:
If it's not set, the value passed is `null`, which makes no difference. I tested it against my vault and it worked as expected.